plugins/events-node/config.d.ts

```diff
@@ -30,5 +30,11 @@ export interface Config {
      * will never be disabled, even if the events backend returns a 404.
      */
     useEventBus?: 'never' | 'always' | 'auto';
+    /**
+     * Timeout in milliseconds for how long to wait for the hanging polling request
+     * on the event bus until triggering a new request. By default, the request will
+     * hang indefinitely until the bus ends the response to signal a new event.
+     */
+    eventBusPollingTimeoutMs?: number;
```

Comment:
I think it'd be ideal to control this from the backend, to keep things simple in a split deployment. The backend uses 55s, but we could make that configurable so that you can bring it down to 25 or so. See https://github.com/backstage/backstage/blob/a096e9ff2d553c7ead7df62794b33babff1d3f37/plugins/events-backend/src/service/hub/createEventBusRouter.ts#L145

What I'm thinking we could to do control this form the backend is to communicate the timeout in a response header, and default to 55s if no header is present.

Beyond that I'm not exactly sure what the effective timeout should be given some known timeout in the backend. I'd imagine it's ideal to typically be closing connections from the frontend and backend first, but not entirely sure which one. My guess would be that letting things close in the backend might produce cleaner logs so we could for example add 1s to the client-side timeout.

Reply:
updated! hopefully I implemented it in line with what you described